test/ibmq/test_experiment.py

```diff
@@ -720,6 +721,22 @@ def test_analysis_results_limit(self):
             self.provider.experiment.analysis_results(limit=-1)
         self.assertIn("limit", str(context_manager.exception))
 
+    def test_experiment_service_prefs(self):
+        """Test accessing and modifying ExperimentService preferences."""
+
+        # Create an experiment service instance
+        service = ExperimentService(self.provider)
+        # Ensure there are more than 0 preferences
+        self.assertGreater(len(service.preferences().keys()), 0)
+        # Get the `auto_save` option
+        auto_save = service.preference('auto_save')
+        # Flip the `auto_save` option
+        service.set_preferences(auto_save=(not auto_save))
+        # Get the `auto_save` pref
+        self.assertEqual(service.preference('auto_save'), not auto_save)
```

Comment:
But this doesn't actually test the preferences are written to / read from `.qiskitrc`, since it just uses what's in the the credentials preferences dictionary.

Reply:
Updated! Uses temp `.qiskitrc` and [verifies changes](https://github.com/Qiskit/qiskit-ibmq-provider/pull/961/files#diff-9275794e4268a2c8dd165487dab5c87c3d271f9692458ac71902381b701725b6R754) with it.